src/GraphQL.MicrosoftDI/DynamicScopedFieldResolver.cs

```diff
@@ -0,0 +1,35 @@
+using GraphQL.Resolvers;
+using Microsoft.Extensions.DependencyInjection;
+
+namespace GraphQL.MicrosoftDI
+{
+    /// <summary>
+    /// When resolving a field, this implementation calls
+    /// <see cref="IResolveFieldContext.RequestServices"/>.<see cref="ServiceProviderServiceExtensions.CreateScope(IServiceProvider)">CreateScope</see>
+    /// to create a dependency injection scope. Then it calls the specified <see cref="IFieldResolver"/>, passing the scoped service provider
+    /// within <see cref="IResolveFieldContext.RequestServices"/>, and returns the result.
+    /// </summary>
+    internal class DynamicScopedFieldResolver : IFieldResolver<Task<object>>
```

Comment:
We already have some similar resolvers. None of them fit?

Reply:
No

Reply:
I can explain, but the bottom line is simply “no”. Especially this one. The most similar code is within the middleware builder.

Reply:
Let me know if you think this should be public.

Reply:
I looked at this again.  Technically any field resolver could use `FuncFieldResolver`.  But then the functionality encapsulated within the field resolver moves to a method somewhere and limits re-use.  In this case I think the better design is to create a dedicated field resolver rather than putting the code directly in the attribute or in a static method.

Reply:
ok